db/db_service.go

```diff
@@ -28,7 +28,17 @@ func NewDBService(db *gorm.DB, eventPublisher events.EventPublisher) *dbService
 	}
 }
 
-func (svc *dbService) CreateApp(name string, pubkey string, maxAmountSat uint64, budgetRenewal string, expiresAt *time.Time, scopes []string, isolated bool, metadata map[string]interface{}) (*App, string, error) {
+func (svc *dbService) CreateApp(
+	name string,
+	pubkey string,
+	maxAmountSat uint64,
+	budgetRenewal string,
+	expiresAt *time.Time,
+	scopes []string,
+	isolated bool,
+	metadata map[string]interface{},
+	walletChildPrivKeyGeneratorFunc func(appId uint32) (string, error),
```

Comment:
Minor: rather than passing this method as an argument, the keys could be passed to the `db_service` in the `NewDBService` function

Reply:
I tried that but get:
```
package command-line-arguments
	imports github.com/getAlby/hub/http
	imports github.com/getAlby/hub/alby
	imports github.com/getAlby/hub/config
	imports github.com/getAlby/hub/db
	imports github.com/getAlby/hub/service/keys
	imports github.com/getAlby/hub/config: import cycle not allowed
```
Is is possible to easily fix this import cycle without too much refactoring?

Reply:
This might be fixed in https://github.com/getAlby/hub/pull/710 which I move the app management out of the db package. This will cause some conflicts here when it's merged, which I can then resolve.

Reply:
fixed!